libminifi/src/c2/C2Agent.cpp

```diff
@@ -556,6 +519,43 @@ void C2Agent::handle_describe(const C2ContentResponse &resp) {
   enqueue_c2_response(std::move(response));
 }
 
+void C2Agent::handle_clear(const C2ContentResponse &resp) {
+  if (resp.name == "connection") {
+    for (const auto& connection : resp.operation_arguments) {
+      logger_->log_debug("Clearing connection %s", connection.second.to_string());
+      update_sink_->clearConnection(connection.second.to_string());
+    }
+  } else if (resp.name == "repositories") {
+    update_sink_->drainRepositories();
+  } else if (resp.name == "corecomponentstate") {
+    for (const auto& corecomponent : resp.operation_arguments) {
+      std::vector<std::shared_ptr<state::StateController>> components = update_sink_->getComponents(corecomponent.second.to_string());
+      auto state_manager_provider = core::ProcessContext::getStateManagerProvider(logger_, controller_, configuration_);
+      if (state_manager_provider != nullptr) {
+        for (auto &component : components) {
+          logger_->log_debug("Clearing state for component %s", component->getComponentName());
+          auto state_manager = state_manager_provider->getCoreComponentStateManager(component->getComponentUUID());
+          if (state_manager != nullptr) {
+            component->stop();
+            state_manager->clear();
+            state_manager->persist();
+            component->start();
+          } else {
+            logger_->log_warn("Failed to get StateManager for component %s", component->getComponentUUID().to_string());
+          }
+        }
+      } else {
+        logger_->log_error("Failed to get StateManagerProvider");
+      }
+    }
+  } else {
+    logger_->log_error("Unknown clear operand %s", resp.name);
+  }
+
+  C2Payload response(Operation::ACKNOWLEDGE, resp.ident, true);
+  enqueue_c2_response(std::move(response));
```

Comment:
Changed to send acknowledge message even for unknown operands as the C2 server should be expecting it.